jedi/evaluate/representation.py

```diff
@@ -557,3 +558,65 @@ def _sub_modules_dict(self):
 
     def py__class__(self):
         return compiled.get_special_object(self.evaluator, 'MODULE_CLASS')
+
+
+class ImplicitNSName(AbstractNameDefinition):
+    """
+    Accessing names for implicit namespace packages should infer to nothing.
+    This object will prevent Jedi from raising exceptions
+    """
+    def __init__(self, implicit_ns_context, string_name):
+        self.implicit_ns_context = implicit_ns_context
+        self.string_name = string_name
+
+    def infer(self):
+        return []
+
+    def get_root_context(self):
+        return self.implicit_ns_context
+
+
+class ImplicitNamespaceContext(use_metaclass(CachedMetaClass, context.TreeContext)):
+    """
+    Provides support for implicit namespace packages
+    """
+    api_type = 'module'
+    parent_context = None
+
+    def __init__(self, evaluator, implicit_ns_info=None):
```

Comment:
I think the only thing left to complain about for me is the nesting of those classes. I would prefer if you just copied the name and paths of the ImplicitNSInfo class over to this one. It's just way more readable IMO. But otherwise, really good job!